pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DistinctFromTransformFunction.java

```diff
@@ -0,0 +1,123 @@
+  // If _distinctType is 1, act as IsDistinctFrom transformation, otherwise act as IsNotDistinctFrom.
+  // This value should only be 0 or 1.
+  private int _distinctType;
+
+  /**
+   * Returns a bit map of corresponding column.
+   * Returns null by default if null option is disabled.
+   */
+  private static @Nullable RoaringBitmap getNullBitMap(ProjectionBlock projectionBlock,
+      TransformFunction transformFunction) {
+    String columnName = ((IdentifierTransformFunction) transformFunction).getColumnName();
+    RoaringBitmap nullBitmap = projectionBlock.getBlockValueSet(columnName).getNullBitmap();
+    return nullBitmap;
+  }
+
+  /**
+   * Fill in results based on functions
+   * @param isDistinctNull return true if one field is null and the other field is not null
+   * @param isBothNull return true if both fields are null
+   * @param length length of the result
+   */
+  private void fillInDistinctNullValue(Function<Integer, Boolean> isDistinctNull, Function<Integer, Boolean> isBothNull,
+      int length) {
+    for (int i = 0; i < length; i++) {
+      if (isDistinctNull.apply(i)) {
+        _results[i] = _distinctType;
+      } else if (isBothNull.apply(i)) {
+        _results[i] = _distinctType ^ 1;
+      }
+    }
+  }
+
+  /**
+   * @param distinctType is set to true for IsDistinctFrom, otherwise it is for IsNotDistinctFrom.
+   */
+  protected DistinctFromTransformFunction(boolean distinctType) {
+    super(distinctType ? TransformFunctionType.NOT_EQUALS : TransformFunctionType.EQUALS);
+    _distinctType = distinctType ? 1 : 0;
+  }
+
+  @Override
+  public String getName() {
+    if (_distinctType == 1) {
+      return TransformFunctionType.IS_DISTINCT_FROM.getName();
+    }
+    return TransformFunctionType.IS_NOT_DISTINCT_FROM.getName();
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    super.init(arguments, dataSourceMap);
+    if (!(_leftTransformFunction instanceof IdentifierTransformFunction)
+        || !(_rightTransformFunction instanceof IdentifierTransformFunction)) {
+      throw new IllegalArgumentException("Only column names are supported in DistinctFrom transformation.");
+    }
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return BOOLEAN_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
+    int length = projectionBlock.getNumDocs();
+    _results = super.transformToIntValuesSV(projectionBlock);
+    RoaringBitmap leftNull = getNullBitMap(projectionBlock, _leftTransformFunction);
+    RoaringBitmap rightNull = getNullBitMap(projectionBlock, _rightTransformFunction);
+    if (leftNull == null && rightNull == null) {
+      return _results;
+    }
+    if (leftNull == null) {
+      fillInDistinctNullValue(rightNull::contains, (Integer i) -> false, length);
```

Comment:
This way won't be as optimized because we still need to call the method.

Reply:
Optimized using forEach loop.